core/math/random.cpp

```diff
@@ -95,6 +95,112 @@ Variant Random::choice(const Variant &p_from) {
 	return Variant();
 }
 
+Variant Random::choices(const Variant &p_from, int p_count, const PoolIntArray &p_weights, bool p_is_cumulative) {
+	int sum = 0;
+	LocalVector<int, int> cumulative_weights;
+	LocalVector<int, int> weights;
+	LocalVector<int, int> indices;
+	Array weighted_choices;
+
+	if((p_from.get_type() == Variant::DICTIONARY) && p_weights.empty()){
+		Dictionary dict = p_from;
+		Array w = dict.values();
+		for (int i = 0; i < w.size(); i++) {
+			weights.push_back(w[i]);
+		}
+	} else {
+		for (int i = 0; i < p_weights.size(); i++) {
+			weights.push_back(p_weights[i]);
+		}
+	}
+
+	if(!weights.empty()){
+		if(p_is_cumulative) {
+			sum = weights[weights.size() - 1];
+			cumulative_weights = weights;
+		} else {
+			for(int i = 0; i < weights.size(); i++) {
+				if(int(weights[i]) < 0) {
+					ERR_FAIL_V_MSG(Array(), "Weights must be positive integers.");
+				} else {
+					sum += (int)weights[i];
+					cumulative_weights.push_back(sum);
+				}
```

Comment:
I think casting is no longer necessary.

```suggestion
				if(weights[i] < 0) {
					ERR_FAIL_V_MSG(Array(), "Weights must be positive integers.");
				} else {
					sum += weights[i];
					cumulative_weights.push_back(sum);
				}
```